tests/integration_test/shadow_system_test.c

```diff
@@ -0,0 +1,917 @@
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file shadow_system_test.c
+ * @brief Integration tests for the SHADOW library via MQTT Library
+ * for the FreeRTOS platform over Secure Sockets as the transport layer.
+ */
+
+/* Standard header includes. */
+#include <string.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <time.h>
+
+
+/* Unity Test framework include. */
+#include "unity_fixture.h"
+
+/* Include connection configurations header. */
+#include "aws_clientcredential.h"
+
+/* Include header for root CA certificates. */
+#include "iot_default_root_certificates.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "core_mqtt.h"
+#include "core_mqtt_state.h"
+
+/* SHADOW API header. */
+#include "shadow.h"
+
+/* Include OpenSSL implementation of transport interface. */
+#include "transport_secure_sockets.h"
+
+/* Include task.h for delay function. */
+#include "task.h"
+
+
+#include "retry_utils.h"
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "TEST"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+#include "logging_stack.h"
+
+/**
+ * These configuration settings are required to run the shadow library integration test.
+ * Throw compilation error if the below configs are not defined.
+ */
+
+#ifndef BROKER_ENDPOINT
+    #define BROKER_ENDPOINT    clientcredentialMQTT_BROKER_ENDPOINT
+#endif
+
+#ifndef BROKER_PORT
+    #define BROKER_PORT    clientcredentialMQTT_BROKER_PORT
+#endif
+
+#ifndef SERVER_ROOT_CA_CERT
+    #define SERVER_ROOT_CA_CERT    tlsATS1_ROOT_CERTIFICATE_PEM
+#endif /* ifndef SERVER_ROOT_CA_CERT_NON_AWS */
+
+/**
+ * @brief Client identifier for MQTT session in the tests.
+ */
+#ifndef TEST_CLIENT_IDENTIFIER
+    #define TEST_CLIENT_IDENTIFIER    clientcredentialIOT_THING_NAME
+#endif
+
+/**
+ * @brief Size of the network buffer for MQTT packets.
+ */
+#define NETWORK_BUFFER_SIZE                 ( 1024U )
+
+/**
+ * @brief Length of the client identifier.
+ */
+#define TEST_CLIENT_IDENTIFIER_LENGTH       ( sizeof( TEST_CLIENT_IDENTIFIER ) - 1u )
+
+/**
+ * @brief Length of the client identifier.
+ */
+#define TEST_CLIENT_IDENTIFIER_LENGTH       ( sizeof( TEST_CLIENT_IDENTIFIER ) - 1u )
```

Comment:
It looks like we have repeat, could you please delete.
```suggestion
```

Reply:
thanks.